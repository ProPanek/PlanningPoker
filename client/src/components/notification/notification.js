import React, {PureComponent} from "react"
import Snackbar from "@material-ui/core/Snackbar"
import IconButton from "@material-ui/core/IconButton"
import CloseIcon from "@material-ui/icons/Close"
import { inject, observer } from "mobx-react"
import styled from "styled-components"
import {decorate, observable} from "mobx";

const NotificationSnackbar = styled(Snackbar)`
  & > div {
    max-width: ${props => props.wide && "100% !important"};
    background-color: ${props => {
      if (props.variant === "error") {
        return "#D32F2F"
      }
      if (props.variant === "warning") {
        return "#FFA001"
      }
      if (props.variant === "info") {
        return "#1876D2"
      }
      if (props.variant === "success") {
        return "#43A047"
      }
    }};
  }
`

class Notification extends PureComponent{

  openNotification = (message, variant) => {
    this.notificationVariant = variant
    this.notificationMessage = message
  }

  componentWillReceiveProps(){
    this.openNotification(this.props.store.notificationMessage, this.props.store.notificationVariant)
  }

  closeNotification =() => {
    this.notificationMessage = null
  }


  render(){
    return (
      <NotificationSnackbar
        anchorOrigin={{
          vertical: "bottom",
          horizontal: "left"
        }}
        wide={300}
        open={!!this.notificationMessage}
        autoHideDuration={3000}
        onClose={this.closeNotification}
        ContentProps={{
          "aria-describedby": "message-id"
        }}
        variant={this.notificationVariant}
        message={<span id="message-id">{this.notificationMessage}</span>}
        action={[
          <IconButton
            key="close"
            aria-label="Close"
            color="inherit"
            onClick={this.closeNotification}
          >
            <CloseIcon name="close" />
          </IconButton>
        ]}
      />
    )
  }
}

decorate(Notification, {
  notificationMessage: observable,
  notificationVariant: observable
});

export default inject("store")(observer(Notification))
