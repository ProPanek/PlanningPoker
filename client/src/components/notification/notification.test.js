import React from "react"
import { shallow } from "enzyme"
import { Notification } from "components/notification/notification"

describe("Notification", () => {
  it("renders without errors", () => {
    const store = {
      uiStore: {
        notificationMessage: "test",
        closeNotification: false,
        notificationOptions: {}
      }
    }
    const component = shallow(<Notification store={store} />)
    expect(component).toMatchSnapshot()
  })
})
