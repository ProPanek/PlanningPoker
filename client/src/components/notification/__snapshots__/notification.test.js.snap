// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders without errors 1`] = `
<Styled(WithStyles(Snackbar))
  ContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <WithStyles(IconButton)
        aria-label="Close"
        color="inherit"
        onClick={false}
    >
        <pure(CloseIcon)
            name="close"
        />
    </WithStyles(IconButton)>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  message={
    <span
      id="message-id"
    >
      test
    </span>
  }
  onClose={false}
  open={true}
  wide={0}
/>
`;
